### PR TITLE
fix: ensure fee explainer initially aligns with trade quote

### LINKED ISSUE #5990: Total fee on the simulate discount page does not change
### Overview

When changing trade amounts the total fee on the simulate discount page does not change.

### References and additional details

1) Log into app.shapeshift and connect to wallet of choice
2) enter a trade that will trigger a fee
3) Note the fee amount then click on it to see the fox power discounts
4) Move to the Simulate discount page and note the total fee amount
5) go back to trade page and enter a new trade amount 
6) repeat above steps and check the simulate fee tab. note the total fee is the same.

https://github.com/shapeshift/web/assets/100157997/42bb2d18-d899-45d0-a424-615a3339cd90

### Acceptance Criteria

Total fee should change when the trade amount changes based on the fox power 

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Ownership

- [X] If my bounty needs engineering or needs product I have added the respective labels on the right
- [X] As the sponsor of this bounty I will review the changes in a preview environment (ops/product) or review the PR (engineering)

### Estimated effort

_No response_

### Sponsor / Stakeholder

_No response_

### Bounty Hunters

- [Join our discord](https://discord.gg/shapeshift)
- Include an expected timeline for you to complete work in the work plan when you apply for this bounty!
- Please refer to [this link](https://shapeshiftdao.gitbook.io/getting-started/how-to-contribute/bounties#resources-for-new-contributors-and-bounty-hunters) for some basic info
- Please ***do not*** start work on this issue until you are approved in Gitcoin.

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Fixes trade fee explainer:
- Reflects active trade quote values and compare against the pre-fox-discounted trade fee, instead of a blanket 30bps
- Pre-populates the sliders for the current quote values
- Formats dollar amounts as user selected currency instead of USD
- Fixes orange fox power slicer triangle to point to the correct part of the slider (it was too far right previously)

Includes general cleanup of fee data around the swappers.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #5990 

## Risk

- Risk that fee amounts are somehow broken or missing in the trade input, trade confirm and trade steps during trade execution
- Risk that fiat amounts are incorrect when on non-usd currencies

## Testing

Test the risk points above to prove no issues. 

### Engineering

Worth testing with monkey patch to ensure big fox holders arent rugged.

Note that swappers only accept integer bps values, so this change take that into account without presenting a confusing ux. It also maintains floating point fox discount amounts for better clarity around exact fox holding, which is also used to compute the fox discount that ends up being presented to the user

## Screenshots (if applicable)

image
image
image

### CODE REVIEW

**Comment by woodenfurniture** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #6010** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6010?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/6010?utm_source=stack-comment).

File: src/components/FeeExplainer/FeeExplainer.tsx [hunk 1]
```diff
@@ -45,8 +46,8 @@ const tradeSizeData = [...Array(CHART_GRANULARITY).keys()].map(
 )
 
 const accessors = {
-  xAccessor: (data: { x: number }) => data.x,
-  yAccessor: (data: { y: number }) => data.y,
```
Comment (woodenfurniture) [inline src/components/FeeExplainer/FeeExplainer.tsx:49]: fixes wall of red when mouse goes over the chart

File: src/lib/fees/model.ts [hunk 2]
```diff
@@ -32,23 +34,12 @@ export const calculateFees: CalculateFeeBps = ({ tradeAmountUsd, foxHeld }) => {
   const midpointUsd = bn(FEE_CURVE_MIDPOINT_USD)
   const feeCurveSteepness = bn(FEE_CURVE_STEEPNESS_K)
 
-  const foxDiscountPercent = BigNumber.minimum(
+  // the following raw values are before the realities of integer bps on-chain
```
Comment (gomesalexandre) [inline src/lib/fees/model.ts:37]: Can we use a different vernacular to make this easier to grasp? E.g something a la "potential"

File: src/lib/fees/model.ts [hunk 3]
```diff
@@ -61,20 +52,44 @@ export const calculateFees: CalculateFeeBps = ({ tradeAmountUsd, foxHeld }) => {
...
-
-  const feeBps = BigNumber.maximum(
-    feeBpsBeforeDiscount.multipliedBy(bn(1).minus(foxDiscountPercent.div(100))),
+  const feeBpsRaw = BigNumber.maximum(
+    feeBpsBeforeDiscountRaw.multipliedBy(bn(1).minus(foxDiscountPercentRaw.div(100))),
     bn(0),
   )
+
+  const feeBpsBeforeDiscount = feeBpsBeforeDiscountRaw.decimalPlaces(0)
+  const feeBpsAfterDiscount = feeBpsRaw.decimalPlaces(0)
```
Comment (gomesalexandre) [inline src/lib/fees/model.ts:61]: Do we want to `dp(0)` this later on, so we have a more accurate discount percentage?

File: src/lib/fees/model.ts [hunk 4]
```diff
@@ -32,23 +34,12 @@ export const calculateFees: CalculateFeeBps = ({ tradeAmountUsd, foxHeld }) => {
...
-      feeUsd: bn(0),
-      feeUsdDiscount: bn(0),
-      foxDiscountPercent,
-      feeUsdBeforeDiscount: bn(0),
-      feeBpsBeforeDiscount: bn(0),
-    }
-  }
-
-  const feeBpsBeforeDiscount = minFeeBps.plus(
+  const feeBpsBeforeDiscountRaw = minFeeBps.plus(
```
Comment (gomesalexandre) [inline src/lib/fees/model.ts:42]: Ditto terminology, not clear what this does unless you read the whole PR

File: src/components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeQuotes.tsx [hunk 5]
```diff
@@ -215,18 +214,14 @@ export const useGetTradeQuotes = () => {
         const receiveAccountNumber = receiveAssetBip44Params?.accountNumber
 
         const tradeAmountUsd = bnOrZero(sellAssetUsdRate).times(debouncedSellAmountCryptoPrecision)
-        const potentialAffiliateBps = DEFAULT_SWAPPER_AFFILIATE_BPS
-        const affiliateBps = (() => {
-          // free trades if there's an error getting foxHeld
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeQuotes.tsx:220]: Shouldn't still be present? This is not a bug, it's part of the flow - failure to get foxHeld means free trades

File: src/components/MultiHopTrade/components/TradeConfirm/ReceiveSummary.tsx [hunk 6]
```diff
@@ -78,6 +80,10 @@ export const ReceiveSummary: FC<ReceiveSummaryProps> = memo(
     const greenColor = useColorModeValue('green.600', 'green.200')
     const textColor = useColorModeValue('gray.800', 'whiteAlpha.900')
 
+    // use the fee data from the actual quote in case it varies from the theoretical calculation
+    const affiliateBps = useAppSelector(selectActiveQuoteAffiliateBps)
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/TradeConfirm/ReceiveSummary.tsx:84]: Won't this rug the "free trades" flow?

File: src/components/FeeExplainer/FeeExplainer.tsx [hunk 7]
```diff
@@ -253,17 +254,32 @@ type FeeOutputProps = {
...
+
+  const userCurrencyToUsdRate = useAppSelector(selectUserCurrencyToUsdRate)
+
+  const feeUserCurrency = useMemo(() => {
+    return feeUsd.times(userCurrencyToUsdRate)
+  }, [feeUsd, userCurrencyToUsdRate])
+
+  const feeUserCurrencyBeforeDiscount = useMemo(() => {
+    return feeUsdBeforeDiscount.times(userCurrencyToUsdRate)
+  }, [feeUsdBeforeDiscount, userCurrencyToUsdRate])
```
Comment (gomesalexandre) [inline src/components/FeeExplainer/FeeExplainer.tsx:271]: Could make it a selector if we wish to

File: src/components/FeeExplainer/FeeExplainer.tsx [hunk 8]
```diff
@@ -272,10 +288,23 @@ export const FeeOutput: React.FC<FeeOutputProps> = ({ tradeSize, foxHolding }) =
         <Flex gap={4}>
           <Box flex={1} textAlign='center'>
             <Text color='text.subtle' translation='foxDiscounts.totalFee' />
-            {feeUsd.lte(0) ? (
+            {feeUserCurrency.lte(0) ? (
```
Comment (gomesalexandre) [inline src/components/FeeExplainer/FeeExplainer.tsx:291]: 💜

File: src/components/MultiHopTrade/components/FeeModal/FeeBreakdown.tsx [hunk 9]
```diff
@@ -1,30 +1,41 @@
...
+
+  const userCurrencyToUsdRate = useAppSelector(selectUserCurrencyToUsdRate)
+
+  const feeBeforeDiscountUserCurrency = useMemo(() => {
+    return feeUsdBeforeDiscount.times(userCurrencyToUsdRate).toFixed(2)
+  }, [feeUsdBeforeDiscount, userCurrencyToUsdRate])
+
+  const feeDiscountUserCurrency = useMemo(() => {
+    return foxDiscountUsd.times(userCurrencyToUsdRate).toFixed(2)
+  }, [foxDiscountUsd, userCurrencyToUsdRate])
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/FeeModal/FeeBreakdown.tsx:34]: Ditto if we wish to

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/FeeStep.tsx [hunk 10]
```diff
@@ -29,32 +26,14 @@ export const FeeStep = ({ isLastStep }: FeeStepProps) => {
   } = useLocaleFormatter()
   const translate = useTranslate()
   const [showFeeModal, setShowFeeModal] = useState(false)
-  const amountAfterDiscountUserCurrency = useAppSelector(selectQuoteAffiliateFeeUserCurrency)
-  const amountBeforeDiscountUserCurrency = useAppSelector(
-    selectPotentialAffiliateFeeAmountUserCurrency,
-  )
-  const potentialAffiliateBps = useAppSelector(selectActiveQuotePotentialAffiliateBps)
+
+  // use the fee data from the actual quote in case it varies from the theoretical calculation
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/FeeStep.tsx:30]: Ditto question WRT free trades

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/FeeStep.tsx [hunk 11]
```diff
@@ -64,23 +43,23 @@ export const FeeStep = ({ isLastStep }: FeeStepProps) => {
   }, [])
 
   const { title, titleProps } = useMemo(() => {
-    return shapeShiftFee && shapeShiftFee.amountAfterDiscountUserCurrency !== '0'
-      ? { title: toFiat(shapeShiftFee.amountAfterDiscountUserCurrency) }
+    return amountAfterDiscountUserCurrency !== '0'
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/FeeStep.tsx:46]: That was my mistake to start with but `.gt(0)` would be better IMO - we should never have negative values, but you never know

File: src/components/FeeExplainer/FeeExplainer.tsx [hunk 12]
```diff
@@ -305,8 +334,9 @@ type FeeExplainerProps = CardProps
 
 export const FeeExplainer: React.FC<FeeExplainerProps> = props => {
   const votingPower = useAppSelector(selectVotingPower)
+  const sellAmountUsd = useAppSelector(selectSellAmountUsd)
 
-  const [tradeSize, setTradeSize] = useState(FEE_CURVE_NO_FEE_THRESHOLD_USD) // default to max below free so we have a value
+  const [tradeSize, setTradeSize] = useState(Number.parseFloat(sellAmountUsd ?? '0'))
```
Comment (gomesalexandre) [inline src/components/FeeExplainer/FeeExplainer.tsx:339]: We probably want to keep this line as-is, and add an effect on `sellAmountUsd` that calls `setTradeSize` if gt(0)

**Review by gomesalexandre — COMMENTED** [review]
Tested locally, LGTM overall minus some conceptual questions about the affiliate bps when missing FOX power, and some issue WRT smolish trades which should be 100% fee-deducted.

Also @woodenfurniture let's remove the `DEFAULT_SWAPPER_AFFILIATE_BPS` constant as it's now effectively superseeded by the FOX discounts curve's `FEE_CURVE_MAX_FEE_BPS`, and effectively not consumed anymore.

## Tested smolly (<1000$) trade against develop:

### Fees breakdown

#### Summary

- Develop

image

- This diff

image

Note trade fee is wrongly 0.00$ (should be the original with potential 30bps), FOX power discount is 0.00$ (should be 100%) 

#### Simulate Discount

 - Develop
 
image

- This diff

image

- Correct trade size is selected instead of default

image

## Tested bigly (1000+$) trade against develop with `vitalik.eth`

### Fees breakdown

- Develop

image
image

- This diff
image

image

Indeed, 29bps is used as a base instead of 30bps. This is indeed correct as we make quotes with 29bps (confirmed against develop).
And indeed, `vitalik.eth` has no FOX power, so there is no discount from the base 29bps

### Simulate Discount

- Develop

image

- This diff

image

The correct fee is now displayed since we do not wrongly consider 29bps to be a discount of 1bps 🎉 

## Tested bigly (1000+$) trade against develop with frame-injected treasury adddress

### Fees breakdown

- Develop

image
image

- This diff

image
image

The correct 100% is still deducted, and we're not starting from a (wrong) base fee of 30bps anymore 🎉 

### Simulate Discount

- Develop

image

- This diff

image

Note current trade size is selected 🎉

**Comment by woodenfurniture** [comment]
@gomesalexandre as discussed I've updated the UI to display the sub $1000 free condition accurately:
image
image
image
image

**Review by gomesalexandre — APPROVED** [review]
Retested locally:

- Confirmed sub-1000-bucks trades are now properly showing a 100% discount

image
image
image

- Larger trades (>1000$) are still happy with full discounts

image
image

- Larger trades (>1000$) are still happy with no discounts

image

Note the wrong dangerous trade message here - not related to this trade and I have seen it randomly with `vitalik.eth` frame-injected wallet for some reason

image

File: src/lib/fees/model.ts (hunk 2 above)
Comment (woodenfurniture) [reply src/lib/fees/model.ts]: great point, actioned with jsdoc comment in 19be7d23c47677edafae86e2457a7bc278d0e430

File: src/lib/fees/model.ts (hunk 4 above)
Comment (woodenfurniture) [reply src/lib/fees/model.ts]: actioned in 19be7d23c47677edafae86e2457a7bc278d0e430

File: src/components/FeeExplainer/FeeExplainer.tsx (hunk 12 above)
Comment (woodenfurniture) [reply src/components/FeeExplainer/FeeExplainer.tsx]: actioned in 19be7d23c47677edafae86e2457a7bc278d0e430v

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/FeeStep.tsx (hunk 11 above)
Comment (woodenfurniture) [reply src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/FeeStep.tsx]: actioned in 19be7d23c47677edafae86e2457a7bc278d0e430

File: src/components/MultiHopTrade/components/TradeConfirm/ReceiveSummary.tsx (hunk 6 above)
Comment (woodenfurniture) [reply src/components/MultiHopTrade/components/TradeConfirm/ReceiveSummary.tsx:84]: solved in 4c28a00c7941ac2e2c4d2320b442c6a89ee3d7fa

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/FeeStep.tsx (hunk 10 above)
Comment (woodenfurniture) [reply src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/FeeStep.tsx:31]: solved in 4c28a00c7941ac2e2c4d2320b442c6a89ee3d7fa

File: src/components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeQuotes.tsx (hunk 5 above)
Comment (woodenfurniture) [reply src/components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeQuotes.tsx:220]: solved in 2d96688c3c4f8e48caed1267aa2734dd8d06eb1b with unit test

**Pull request merged**